common/com/twitter/intellij/pants/model/PantsOptions.java

```diff
@@ -66,11 +66,11 @@ public boolean supportsLint() {
 
   public static PantsOptions getPantsOptions(@NotNull final String pantsExecutable) {
     File pantsExecutableFile = new File(pantsExecutable);
-    PantsOptions cache = optionsCache.get(pantsExecutableFile);
-    if (cache != null) {
-      return cache;
-    }
+    return optionsCache.computeIfAbsent(pantsExecutableFile, file -> execPantsOptions(pantsExecutable, file));
```

Comment:
From the `computeIfAbsent` docstring

> Some attempted update operations on this map by other threads may be blocked while computation is in progress, so the computation should be **short and simple**, and must not attempt to update any other mappings of this map.

Given that executing a Pants command is not "short and simple", would it make sense to use another synchronization mechanism?

Reply:
The cause for this suggestion states that:
> **Some attempted update operations on this map by other threads may be blocked while computation is in progress**, so the computation should be short and simple, and must not attempt to update any other mappings of this map.

Other simple solution for solving this issue would be to synchronize entire method, which would effectively block all attempted updates instead of just some of them. 

Given that our goal here is to initialize the cache and it is only done once per pants executable, I think the risk of congestion is quite low. Nevertheless thanks for pointing that out - I will add a note in the code so that, if this becomes an issue, it is easier to track and fix.